### PR TITLE
Update to latest revery

### PR DESCRIPTION
**Pull request by glennsl**
Something's off about the initialization of pipes, as they appear right in front of the player instead of off the right of the screen. Otherwise it seems to work.

Also, in hindsight, instead of rewriting all uses of `Rectangle` to `BoundingBox2d`, I probably should have just made a polyfill `Rectangle` module.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,23 +7,28 @@
...
+    //"@opam/js_of_ocaml": "github:ocsigen/js_of_ocaml:js_of_ocaml.opam#db257ce",
+    //"@opam/js_of_ocaml-compiler": "github:ocsigen/js_of_ocaml:js_of_ocaml-compiler.opam#db257ce",
+    "@brisk/brisk-reconciler": "briskml/brisk-reconciler#0a2e476",
+    "esy-cmake": "prometheansacrifice/esy-cmake#2a47392def755",
+    "esy-sdl2": "revery-ui/esy-sdl2#b63892b",
+    "esy-skia": "revery-ui/esy-skia#91b10c9",
+    "rebez": "github:jchavarri/rebez#03fa3b7",
+    "revery": "revery-ui/revery#8417070",
+    "timber": "glennsl/timber#ae065bb",
+    "reason-harfbuzz": "revery-ui/reason-harfbuzz#0848520"
```
Comment (glennsl) [inline package.json]: I'm not sure all of these resolutuons are necessary, but some of them definitely are. `timber` hasn't been released to OPAM at all (I should do something about that), and I think the `reason-harfbuzz` release at least is outdated.

Having these don't give the best impression, so we should try to make sure we have a release of revery that doesn't need any of them.

**Comment by glennsl** [comment]
Should be good now. Thanks for finding the bug @bryphe!

**Review by bryphe — APPROVED** [review]
Thanks for updating this and bringing back the `Rectangle`! 🎉

File: package.json (hunk 1 above)
Comment (bryphe) [reply package.json]: Definitely - just published `reason-harfbuzz@1.91.5004` so hopefully we can use that.

For Revery, we can publish a new version prior to the workshop.
For Timber, if you put that on OPAM or NPM, we should be set.

Regarding the other dependencies:
- `brisk-reconciler` - not needed; direct dependency specified by Revery
- `esy-sdl2` - not needed
- `esy-skia` - not needed
- `rebez` - not needed

The only tricky one is `esy-cmake`... I'm not sure how to address that one. There's a few nested dependencies that need it. We could publish a namespaced package like `@revery-ui/cmake` and fix the chain of dependencies for that.

Comment (glennsl) [reply package.json]: Thanks! I updated revery to latest and got rid of all the unnecessary resolutions. Looks much better now! We can sort out the rest later.

**Pull request merged**